administrator/components/com_joomgallery/controllers/comments.php

```diff
@@ -117,6 +117,15 @@ public function approve()
    */
   public function remove()
   {
+    // Check whether we are allowed to delete
+    $canDo = JoomHelper::getActions();
+    if(!$canDo->get('core.delete'))
+    {
+      $this->setRedirect($this->_ambit->getRedirectUrl('maintenance&tab=comments'), JText::_('JLIB_RULES_NOT_ALLOWED'), 'error');
```

Comment:
It is not correct to **always** redirect to the maintenance manager if the authorization is not available. Note that it is also possible to delete comments in the comment manager, then you should redirect to the comment manager if the authorization is not available.

Reply:
Thanks for the hint. I commited a solution to this problem.

Reply:
When deleting comments in the comment manager, you are still redirected to the maintenance manager.

Reply:
You are right. Should be fixed now...